crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -907,6 +907,48 @@ async fn test_dev_inspect_gas_coin_argument() {
     assert!(return_values.is_empty());
 }
 
+#[tokio::test]
+async fn test_dev_inspect_gas_price() {
+    let (_, fullnode, _object_basics) =
+        init_state_with_ids_and_object_basics_with_fullnode(vec![]).await;
+
+    let sender = SuiAddress::random_for_testing_only();
+    let recipient = SuiAddress::random_for_testing_only();
+    let amount = 500;
+    let pt = {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        builder.pay_sui(vec![recipient], vec![amount]).unwrap();
+        builder.finish()
+    };
+    let kind = TransactionKind::programmable(pt);
+    let error = fullnode
+        .dev_inspect_transaction_block(sender, kind.clone(), Some(1))
+        .await
+        .unwrap_err();
+    assert!(
+        matches!(
+            UserInputError::try_from(error.clone()).unwrap(),
+            UserInputError::GasPriceUnderRGP { .. }
```

Comment:
nit: Is there a way for us to set the RGP within the test so that it's obvious that the gas price set by dev inspect is too low?

Alternatively, we could set the price to RGP - 1, as I see below there's a way to get the RGP from a fullnode for testing purposes.